db/cache/charts.go

```diff
@@ -302,22 +303,54 @@ func newWindowSet(size int) *windowSet {
 	}
 }
 
+// missedSet is for missed votes data that is grouped at an interval of 144
+// blocks on mainnet. missed votes txs don't appear till block 4096 on mainnet
+// when the first POS votes txs were made.
+type missedSet struct {
+	cacheID uint64
+	Count   ChartUints
+	Time    ChartUints
+	Height  ChartUints
+}
+
+// Snip truncates the missedSet to a provided length.
+func (set *missedSet) Snip(length int) {
+	if length < 0 {
+		length = 0
+	}
+	set.Count = set.Count.snip(length)
+	set.Time = set.Time.snip(length)
+	set.Height = set.Height.snip(length)
+}
+
+// Constructor for a sized missedSet.
+func newMissedSet(size int) *missedSet {
+	return &missedSet{
+		Count:  newChartUints(size),
+		Time:   newChartUints(size),
+		Height: newChartUints(size),
+	}
+}
+
```

Comment:
Why not use the `windowSet`?

Reply:
Because missed votes do not appear till after the stake validation height.
I had made a try to match the `windowSet` data with missed votes but I experienced some slice length inconsistencies. I will make another try.